extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.research.extractMethod
+
+import org.apache.logging.log4j.LogManager
+import org.jetbrains.research.extractMethod.core.extractors.RefactoringsExtractor
+import org.jetbrains.research.pluginUtilities.openRepository.getKotlinJavaRepositoryOpener
+import org.jetbrains.research.pluginUtilities.preprocessing.getKotlinJavaPreprocessorManager
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.streams.toList
+
+class BaseRunner {
+    private val logger = LogManager.getLogger(BaseRunner::class)
+    private val preprocessor = getKotlinJavaPreprocessorManager(null)
+    private val repositoryOpener = getKotlinJavaRepositoryOpener()
+
+    private fun getSubdirectories(path: Path): List<Path> {
+        return Files.walk(path, 1)
+            .filter { Files.isDirectory(it) && !it.equals(path) }
+            .toList()
+    }
+
+    fun runExtractions(inputDir: Path, extractor: RefactoringsExtractor) {

```

Comment:
please, try to do it in another way ([standardRepositoryOpener](https://github.com/JetBrains-Research/Kotlin-Analysis/blob/5af4060d8be0ef4dc8f79e2beee78a44a4042cad/kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/util/RepositoryOpenerUtil.kt#L41))